github-workitem/src/main/java/org/jbpm/process/workitem/github/FetchIssuesWorkitemHandler.java

```diff
@@ -0,0 +1,120 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Wid(widfile = "GithubFetchIssues.wid", name = "GithubFetchIssues",
+        displayName = "GithubFetchIssues",
+        defaultHandler = "mvel: new org.jbpm.process.workitem.github.FetchIssuesWorkitemHandler()",
+        documentation = "${artifactId}/index.html",
+        parameters = {
+                @WidParameter(name = "User", required = true),
+                @WidParameter(name = "RepoName", required = true),
+                @WidParameter(name = "IssuesState")
+        },
+        results = {
+                @WidResult(name = "IssuesList")
+        },
+        mavenDepends = {
+                @WidMavenDepends(group = "${groupId}", artifact = "${artifactId}", version = "${version}")
+        },
+        serviceInfo = @WidService(category = "${name}", description = "${description}",
+                keywords = "github,repo,repository,fetch,issues",
+                action = @WidAction(title = "Fetch issues for a project from GitHub")
+        ))
+public class FetchIssuesWorkitemHandler extends AbstractLogOrThrowWorkItemHandler {
+
+    private String userName;
+    private String password;
+    private GithubAuth auth = new GithubAuth();
+
+    private static final Logger logger = LoggerFactory.getLogger(FetchIssuesWorkitemHandler.class);
+    private static final String RESULTS_VALUE = "IssuesList";
+
+    public FetchIssuesWorkitemHandler(String userName,
+                                      String password) {
+        this.userName = userName;
+        this.password = password;
+    }
+
+    public void executeWorkItem(WorkItem workItem,
+                                WorkItemManager workItemManager) {
+        try {
+            RequiredParameterValidator.validate(this.getClass(),
+                                                workItem);
+
+            Map<String, Object> results = new HashMap<String, Object>();
+
+            String user = (String) workItem.getParameter("User");
+            String repoName = (String) workItem.getParameter("RepoName");
+            String issuesState = (String) workItem.getParameter("IssuesState");
+
+            IssueService issueService = auth.getIssueService(this.userName,
+                                                             this.password);
+
+            // default to open
+            if (issuesState == null || (!issuesState.equalsIgnoreCase("open") || !issuesState.equalsIgnoreCase("closed"))) {
+                issuesState = IssueService.STATE_OPEN;
+            }
+
+            List<Issue> issues = issueService.getIssues(user,
+                                                        repoName,
+                                                        Collections.singletonMap(IssueService.FILTER_STATE,
+                                                                                 issuesState.toLowerCase()));
+
+            // no issues is acceptable
+            if (issues != null && issues.size() > -1) {
```

Comment:
Size will be always > -1 so I think the second part of the condition is not needed.

Reply:
oops my bad - updated